controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/SystemApplication.java

```diff
@@ -0,0 +1,72 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.controller.application;
+
+import com.yahoo.component.Version;
+import com.yahoo.config.provision.ApplicationId;
+import com.yahoo.config.provision.NodeType;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * This represents a system-level application in hosted Vespa. E.g. the zone-application.
+ *
+ * @author mpolden
+ */
+public class SystemApplication {
```

Comment:
I'd prefer an enum, and passing the `Version` as a separate argument to the `deploy(...)` variants, but not a strong opinion on this.

Reply:
Agreed.